lib/engine/step/g_1867/merge.rb

```diff
@@ -92,14 +101,100 @@ def move_assets(from, to)
         def move_tokens(from, to)
           from.tokens.each do |token|
             new_token = to.next_token
-            token.swap!(new_token)
+            unless new_token
+              new_token = Engine::Token.new(to)
+              to.tokens << new_token
+            end
+
+            city = token.city
+            token.remove!
+            city.place_token(to, new_token, check_tokenable: false)
           end
         end
 
+        def finish_merge
+          @game.log << "Finish merge of #{@merging.map(&:name).join(',')}"
+          @merge_major = true
+        end
+
+        def finish_merge_to_major(action)
+          target = action.corporation
+          initiator = action.entity.owner
+          # PAR price is average of lowest and highest priced
+          # rounded down between 100-200 (200 can be ignored since max price of minor) between
+          min = @merging.map { |c| c.share_price.price }.min
+          max = @merging.map { |c| c.share_price.price }.max
+          merged = [100, (max + min) / 2].max
+
+          @game.stock_market.set_par(target, @game.find_share_price(merged))
+
+          # Replace the entity with the new one.
+          @round.entities[@round.entity_index] = target
+
+          @merge_major = false
+          owners = []
+          # @todo: sort merging around the initiator then table order
+          @merging.each do |corporation|
+            owner = corporation.owner
+            owners << owner
+            @game.close_corporation(corporation)
+
+            share = target.shares.last
+
+            if target.shares.first.president && owner.percent_of(target) == 10
+              # give the 10% back
+              presidency = target.shares.first
+              @game.share_pool.move_share(owner.shares_of(target).first, presidency.owner)
```

Comment:
Seems there are a number of ways to do this. In 18CO, I used `share.transfer(...)`. When should we use `@game.share_pool` instead?

Reply:
not sure, can both accomplish the same thing?

Reply:
i can rework this later, i'll need to do the later on the case that people are buying shares when they're at 10% anyway

Reply:
I remember, I added it for 1817 to avoid making that public :)